File: gitdot-core/src/dto/domain/ci/runner.rs
mod create_runner;
mod create_runner_token;
mod delete_runner;
mod get_runner;
mod list_runners;
mod verify_runner;

use chrono::{DateTime, Utc};
use uuid::Uuid;

use crate::model::{Runner, RunnerOwnerType};

pub use create_runner::CreateRunnerRequest;
pub use create_runner_token::{CreateRunnerTokenRequest, CreateRunnerTokenResponse};
pub use delete_runner::DeleteRunnerRequest;
pub use get_runner::GetRunnerRequest;
pub use list_runners::ListRunnersRequest;
pub use verify_runner::VerifyRunnerRequest;

#[derive(Debug, Clone)]
pub struct RunnerResponse {
    pub id: Uuid,
    pub name: String,
    pub owner_id: Uuid,
    pub owner_name: String,
    pub owner_type: RunnerOwnerType,
    pub last_active: Option<DateTime<Utc>>,
    pub created_at: DateTime<Utc>,
}

impl From<Runner> for RunnerResponse {
    fn from(runner: Runner) -> Self {
        Self {
            id: runner.id,
            name: runner.name,
            owner_id: runner.owner_id,
            owner_name: runner.owner_name,
            owner_type: runner.owner_type,
            last_active: runner.last_active,
            created_at: runner.created_at,
        }
    }
}

pub type CreateRunnerResponse = RunnerResponse;
pub type GetRunnerResponse = RunnerResponse;

File: gitdot-web/packages/gitdot-api/src/resource/runner.ts
import { z } from "zod";

export const RunnerResource = z.object({
  id: z.uuid(),
  name: z.string(),
  owner_id: z.uuid(),
  owner_name: z.string(),
  owner_type: z.string(),
  last_active: z.iso.datetime().nullable(),
  created_at: z.iso.datetime(),
});
export type RunnerResource = z.infer<typeof RunnerResource>;

export const RunnerTokenResource = z.object({
  token: z.string(),
});
export type RunnerTokenResource = z.infer<typeof RunnerTokenResource>;
